### PR TITLE
done ts_4

### PR DESCRIPTION
**Pull request by hellhellpnick**

### CODE REVIEW

**Review by Korew — CHANGES_REQUESTED** [review]
Please create functionality for adding/removing data from db.json

Separate input for ID field not needed - all id's should be created in background. 

When I click on delete button, I assume row will be deleted (will be good if you create confirmation popup to avoid accidental deletion).

For mass actions (when you select several checkboxes) - you need another button.

Please read comments and fix them all over the project (not only where I added comment)

If you repeat code twice (or more) - please create function for it.

File: homework_4/css/form-add.css [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+.form-add-input {
+  border: 1px solid var(--color-text-ligth);
+  padding: 10px;
+  outline: none;
+  color: var(--color-text);
+
+  margin-bottom: 20px;
+}
+
+.form-add-btn {
```
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework_4/css/form-add.css:34]: Fixed

File: homework_4/css/form-add.css [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+.form-add {
+  display: flex;
+  flex-direction: column;
+}
+
+.form-add-input {
+  border: 1px solid var(--color-text-ligth);
+  padding: 10px;
+  outline: none;
```
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework_4/css/form-add.css:28]: Fixed

File: homework_4/css/inherit.css [hunk 3]
```diff
@@ -0,0 +1,5 @@
+.container {
+  max-width: var(--container-width);
+  padding: 0 20px;
+  margin: 0 auto;
+}
```
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework_4/css/inherit.css:5]: Fixed

File: homework_4/js/main.js [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+        dataCity = td[4].innerHTML;
+        dataPhone = td[2].innerHTML;
+        dataCompany = td[1].innerHTML;
+        dataCountry = td[5].innerHTML;
+        dataName = td[7].innerHTML;
+        dataEmail = td[6].innerHTML;
+      }
+    });
+  }
+  let box = document.createElement('div');
```
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework_4/js/main.js:26]: Fixed

File: homework_4/js/main.js [hunk 5]
```diff
@@ -0,0 +1,327 @@
+const btnAddElement = document.querySelector('.form-btn-add');
+const tableList = document.getElementById('table-list');
+
+const workWithElemnt = function workWithElemntToMainTable(id) {
```
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework_4/js/main.js:4]: Fixed

File: homework_4/js/main.js [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+    inputAddress.value = dataAddress;
+    inputCity.value = dataCity;
+    inputCompany.value = dataCompany;
+    inputCountry.value = dataCountry;
+    inputEmail.value = dataEmail;
+    inputName.value = dataName;
+    inputPhone.value = dataPhone;
+  }
+
+  btnClose.setAttribute('type', 'submit');
```
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework_4/js/main.js:119]: Fixed

File: homework_4/js/main.js [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+  form.appendChild(btnClose);
+
+  box.appendChild(formTitle);
+  box.classList.add('position-center');
+  box.appendChild(form);
+  document.getElementsByTagName('body')[0].appendChild(box);
+
+  btnClose.addEventListener('click', () => box.remove());
+
+  form.addEventListener('submit', (e) => {
```
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework_4/js/main.js:159]: I know, but it's making my `prettier`

File: homework_4/js/main.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework_4/js/main.js:4]: When I was at work, I was advised to use the entire function declaration. First, it is called a short name for use, and if you do not understand what this function does (by the short name), then you look at the full name

[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework_4/js/main.js:4]: and in any case, you are called through a short name

File: homework_4/js/main.js [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+
+const workWithElemnt = function workWithElemntToMainTable(id) {
+  let dataId, dataAddress, dataCity, dataCompany, dataCountry, dataEmail, dataName, dataPhone;
+
+  let td;
+  const tableBody = document.querySelectorAll('tr');
+
+  if (typeof id !== 'undefined') {
+    tableBody.forEach((elem) => {
+      if (id == elem.id) {
```
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework_4/js/main.js:12]: Fixed

File: homework_4/js/main.js [hunk 9]
```diff
@@ -0,0 +1,327 @@
...
+  let labelCompany = document.createElement('label');
+  let inputCountry = document.createElement('input');
+  let labelCountry = document.createElement('label');
+  let inputEmail = document.createElement('input');
+  let labelEmail = document.createElement('label');
+  let inputName = document.createElement('input');
+  let labelName = document.createElement('label');
+  let inputPhone = document.createElement('input');
+  let labelPhone = document.createElement('label');
+  let inputId = document.createElement('input');
```
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework_4/js/main.js:44]: Okay. I'm understand what do you mean, but how should I get id from new elements when I create?

File: homework_4/js/main.js [hunk 10]
```diff
@@ -0,0 +1,327 @@
...
+  inputPhone.setAttribute('name', 'userPhone');
+  inputPhone.setAttribute('required', 'required');
+  inputPhone.setAttribute('minlength', '4');
+  inputPhone.classList.add('form-add-input');
+
+  inputId.setAttribute('type', 'tel');
+  inputId.setAttribute('name', 'userId');
+  inputId.setAttribute('readonly', 'readonly');
+  inputId.setAttribute('value', `${id ? dataId : createRandomSymbols(24)}`);
+  inputId.classList.add('form-add-input');
```
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework_4/js/main.js:107]: Fixed

File: homework_4/js/main.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (Korew) [reply homework_4/js/main.js:202]: ok, not a problem

File: homework_4/js/main.js [hunk 11]
```diff
@@ -0,0 +1,442 @@
...
+    } else {
+      addElement(obj);
+    }
+    addElementOrEditToFile(obj);
+    box.remove();
+    body.classList.toggle('is-dont-point-event');
+  });
+};
+
+const createRandomSymbols = function createRandomCountSymbols(length) {
```
Comment (Korew) [inline homework_4/js/main.js:235]: please fix

File: homework_4/js/main.js [hunk 12]
```diff
@@ -0,0 +1,442 @@
...
+  const labelCompany = document.createElement('label');
+  const inputCountry = document.createElement('input');
+  const labelCountry = document.createElement('label');
+  const inputEmail = document.createElement('input');
+  const labelEmail = document.createElement('label');
+  const inputName = document.createElement('input');
+  const labelName = document.createElement('label');
+  const inputPhone = document.createElement('input');
+  const labelPhone = document.createElement('label');
+  const inputId = document.createElement('input');
```
Comment (Korew) [inline homework_4/js/main.js:123]: remove input for id

**Review by Korew — CHANGES_REQUESTED** [review]

**Review by Korew — APPROVED** [review]

**Pull request merged**